src/main/resources/com/nec/cyclone/cpp/cyclone/packed_transfer.cc

```diff
@@ -138,16 +134,14 @@ void merge_scalar_transfer(size_t batch_count, size_t total_element_count, char*
     cur_out_data_pos += col_in->data_size;
 
     //std::cout << "merge_scalar_transfer: copy validity" << std::endl;
-    if(b == 0){
-      // just copy over the given validity buffer for the first batch
-      std::memcpy(out_validity_buffer, &input_data[cur_data_pos], col_in->validity_buffer_size);
-      i = col_in->element_count;
-    }else{
-      uint64_t* batch_validity_buffer = reinterpret_cast<uint64_t *>(&input_data[cur_data_pos]);
-      #pragma _NEC novector
-      for(auto ci = 0; ci < col_in->element_count; ci++){
-        set_valid_bit(out_validity_buffer, i++, get_valid_bit(batch_validity_buffer, ci));
-      }
+    uint64_t* batch_validity_buffer = reinterpret_cast<uint64_t *>(&input_data[cur_data_pos]);
+    dangling_bits = cyclone::append_bitsets(
+        &out_validity_buffer[cur_out_validity_data_pos], dangling_bits,
+        batch_validity_buffer, col_in->element_count);
+    cur_out_validity_data_pos += frovedis::ceil_div(col_in->element_count, size_t(64));
+    if(dangling_bits > 0){
+      // last part is not entirely filled yet
+      cur_out_validity_data_pos -= 1;
```

Comment:
Perhaps I might be misunderstanding here - `dangling_bits` is set to `second_bit_count % bits_per_T;` inside `cyclone::bitsets()`.  But if `cur_out_validity_data_pos` is subtracted by 1, will that cause those remainder bits that occupy the first N positions in the last `T` byte-block to be overwritten by `append_bitsets` in the next loop iteration?

Reply:
It should be overwriting the "unused" bits within that block on the next iteration.